### PR TITLE
planning: fix a bug in destination. when stop point is not on lane, s…

### PR DESCRIPTION
**Pull request by jmtao**
…till need create the virtual obstacle, but skip the STOP decision

### CODE REVIEW

File: modules/planning/common/util/common.cc [hunk 1]
```diff
@@ -108,6 +108,12 @@ int BuildStopDecision(const std::string& stop_wall_id,
     return -1;
   }
 
+  const auto stop_wall_box = stop_wall->PerceptionBoundingBox();
```
Comment (YajiaZhang) [inline modules/planning/common/util/common.cc]: const auto& ?

File: modules/planning/traffic_rules/destination.cc [hunk 2]
```diff
@@ -78,16 +54,32 @@ int Destination::MakeDecisions(
...
+  const auto& adc_sl = reference_line_info->AdcSlBoundary();
+  const auto& dest =
+      PlanningContext::Instance()->mutable_planning_status()->destination();
+  if (adc_sl.start_s() > dest_sl.s() && !dest.has_passed_destination()) {
+    ADEBUG << "Destination at back, but we have not reached destination yet";
+    return 0;
+  }
+
+  double dest_lane_s = std::max(
+      0.0, routing_end.s() - FLAGS_virtual_stop_wall_length -
```
Comment (YajiaZhang) [inline modules/planning/traffic_rules/destination.cc]: Better to use fmax for floating numbers.

File: modules/planning/common/util/common.cc (hunk 1 above)
Comment (jmtao) [reply modules/planning/common/util/common.cc]: fixed. amended. thx~

File: modules/planning/traffic_rules/destination.cc (hunk 2 above)
Comment (jmtao) [reply modules/planning/traffic_rules/destination.cc]: fixed. amended. thx~

**Pull request merged**